### PR TITLE
Theme Support with Structure Updates

### PR DESCRIPTION
**Pull request by bejzik8**
# Summary

-Updated widget presentation conditions
-Created `useNimi` fetching hook
-Updated structure of `App` component
-Created `NimiCard` wrapper component
-Separated `Header` component from `NimiComponent`
-Restructured `ThemeProvider`
-Updated `StyledWrapper` styling
-Removed unused styles
-Created `NimiTheme` together with all necessary classes
-Fixed the issue with `key`s while mapping links
-Fixed the issue with `key`s while mapping addresses
-Updated whole application according to the new changes
-Updated global styles
-Created `Header` component
-Created `NimiLink` component
-Created `useLinks` service
-Created `LinksSection` component
-Updated `POAPWidget` component
-Created `ProfilePhoto` component
-Created `UserInfo` component
-Added animated background image
-Created `WidgetsSection` component
-Created `BlockchainAddresses` component
-Created `Address` component
-Created `NimiSection` component
-Created **Devcon Nimi Theme**
-Created `Footer` component
-Created `themeScheme`
-Updated `POAPWidget` context scheme
-Added `share-new.svg` and `qr.svg` assets
-Created new ENS information components
-Removed old `ShareButton` component
-Removed bunch of unused stuff
-Removed `isLanding` state
-Removed `isPreview` state.

# Background

Creating features that will support theming and improving application's structure.

### CODE REVIEW

File: src/library/types/NimiThemeDraft.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+export class NimiThemeFooter {
+  backgroundImage: string;
+
+  constructor(backgroundImage: string) {
+    this.backgroundImage = backgroundImage;
+  }
+}
+
+export class NimiTheme {
```
Comment (adamazad) [inline src/library/types/NimiThemeDraft.ts:60]: How do you serialize this to move around?

File: src/library/types/NimiStandard.ts [hunk 2]
```diff
@@ -42,4 +42,6 @@ export interface Nimi {
    * Supported widgets
    */
   widgets: NimiWidget[];
+
+  theme?: any;
```
Comment (adamazad) [inline src/library/types/NimiStandard.ts:46]: Not very strong, I imagine this can't be an object

File: src/library/services/useLink.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    if (isValidUrl(content)) {
+      const url = new URL(content);
+      return url.pathname.replace(new RegExp('/', 'g'), '');
+    }
+
+    return content;
+  }
+
+  function generateLink(link: NimiLinkBaseDetails): string {
```
Comment (adamazad) [inline src/library/services/useLink.ts:24]: the name is confusing

File: src/library/components/BlockchainAddressesSection/BlockchainAddressesSection.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    navigator.clipboard.writeText(address);
+    open(feedback);
+  };
+
+  return (
+    <NimiSection marginBottom="25px">
+      <SectionItemContainer>
+        {addresses.map(({ address, blockchain }) => (
+          <Address
+            key={`${blockchain}-${address}-${nanoid()}`}
```
[the remark this answers is not in the document]
Comment (bejzik8) [reply src/library/components/BlockchainAddressesSection/BlockchainAddressesSection.tsx:23]: Agree. 👍🏼

File: src/library/components/Footer/Footer.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      );
+    default:
+      return (
+        <>
+          <NimiLogo />
+          <NimiText />
+        </>
+      );
+  }
+}
```
[the remark this answers is not in the document]
Comment (bejzik8) [reply src/library/components/Footer/Footer.tsx:23]: I know, I talked to @zamli, I don't know if you are in sync, we will create 5 themes for **Devcon** and after that he'll try to standardise the theme properties.

File: src/library/components/Header/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+      <Logo onClick={() => window.open('https://nimi.link/', '_blank')} />
+    </Container>
+  );
+}
+
+function getHeaderBackground(themeType: NimiThemeType) {
+  switch (themeType) {
+    case NimiThemeType.DEVCON:
+      return css`
+        background: url(https://cdn.discordapp.com/attachments/1018317760968806411/1026246674793910372/clouds_1.png);
```
[the remark this answers is not in the document]
Comment (bejzik8) [reply src/library/components/Header/Header.tsx:36]: I also talked to @zamli about this, he said we should be fine for now but I agree that needs to be updated at some point.

File: src/library/components/LinksSection/LinksSection.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      open(`${link.type.charAt(0).toUpperCase() + link.type.slice(1).toLowerCase()} copied to the clipboard!`);
+    } else {
+      window.open(generateLink(link), '_blank');
+    }
+  };
+
+  return (
+    <Container>
+      {links.map((link) => (
+        <NimiLink key={`${link.type}-${nanoid()}`} onClick={() => handleLinkClick(link)}>
```
[the remark this answers is not in the document]
Comment (bejzik8) [reply src/library/components/LinksSection/LinksSection.tsx:31]: Agree but we need a solution for this. I updated it now but I think `linkId` should be saved with the other data to **JSON** file. Let me know what do you think, I added link's Id as an optional property and it is being created in `Nimi App`.

File: src/library/services/useLink.ts (hunk 3 above)
Comment (bejzik8) [reply src/library/services/useLink.ts:24]: It is the old functionality, I just moved it to the service.

File: src/library/components/BlockchainAddressesSection/BlockchainAddressesSection.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (bejzik8) [reply src/library/components/BlockchainAddressesSection/BlockchainAddressesSection.tsx:23]: Commented about it up. Resolved!

File: src/library/components/NimiComponent/NimiComponent.tsx [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+  const [profilePhotoRotated, setProfilePhotoRotated] = useState(false);
+
+  const validatedNimi = nimiValidator.validateSync(nimi) as unknown as Nimi;
+  const { ensAddress, displayName, image, addresses, description, ensName, links, widgets, theme } = validatedNimi;
+
+  return (
+    <NimiContainer>
+      <Header themeType={theme.type} />
+      <Content>
+        <ProfilePhoto ensName={ensName} image={image} theme={theme} profilePhotoRotated={profilePhotoRotated} />
```
[the remark this answers is not in the document]
Comment (bejzik8) [reply src/library/components/NimiComponent/NimiComponent.tsx:31]: I do not understand...

**Pull request merged**